src/Processors/Transforms/DistinctSortedTransform.h

```diff
@@ -57,7 +58,8 @@ class DistinctSortedTransform : public ISimpleTransform
     };
     PreviousChunk prev_chunk;
 
-    Names columns_names;
+    Names column_names;
+    mutable ColumnNumbers column_positions;
```

Comment:
Why don't set it in the constructor from the header?

Reply:
I did it this way at first (https://github.com/ClickHouse/ClickHouse/commit/416c14afd8c48483f118e3585a8cfe426f03e2ba, basically copy-paste from `DistinctTransform.cpp`), but a bit wrong and got confused by header with 0 columns.
Just recheck and see that `header_` is used in constructor, which is already moved into member`header`(unlike in `DistinctTransform`). So, the PR is back to calculation column positions in constructor.